### ISSUE TITLE
Move on from deprecated pkg_resources api

### ISSUE THREAD

**Issue by farhan** [openedx/platform-plugin-aspects#81]
Shift the code from `pkg_resources` to `importlib.resources` api

Migration guide: https://importlib-resources.readthedocs.io/en/latest/migration.html
Decision making story: https://github.com/openedx/XBlock/issues/676
Reference PR to take learning from: https://github.com/openedx/xblock-sdk/pull/350/files

**Prerequisites:**
- Support of python 3.8 should be dropped. [Relevant epic](https://github.com/openedx/public-engineering/issues/274)

**Acceptance Criteria:**
- Make sure the repository is using xblock version 5.0.0 or above. Upgrade the version if required, preferably via the make upgrade command.
- There should be no usage of `pkg_resources` in the repository
- Test cases should be passed
- XBlock should perform all of its functionality related to change (loading resources)

**Comment by bmtcril** [comment]
This is another one that's blocked until we do our Aspects v2 breaking changes, until then we still need to support 3.8 for named releases back to Nutmeg.

**Comment by feanil** [comment]
@bmtcril is this still blocked or can we start work on it?

**Comment by bmtcril** [comment]
@feanil yep, it's part of the breaking changes for Aspects 2.0 that I pulled together for scheduling yesterday: https://github.com/orgs/openedx/projects/5/views/12 We should be starting on those tasks this week or next.
